### PR TITLE
[MXNET-46]remove set CMAKE_GENERATOR_TOOLSET

### PR DESCRIPTION
**Pull request by yajiedesign**
## Description ##
remove set CMAKE_GENERATOR_TOOLSET,This parameter should be set up by -T

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:

### CODE REVIEW

**Comment by cjolivier01** [comment]
now tim build with CUDA, you must specify a -T option?

**Comment by cjolivier01** [comment]
to build*

**Comment by yajiedesign** [comment]
@cjolivier01 If want to choose a CUDA version, you need to set it with -T.like` cuda=8.0`.
if not set,cmake will automatically select the latest version.

**Comment by cjolivier01** [comment]
it looks like USE_OLDCMAKECUDA is removed?

**Comment by yajiedesign** [comment]
yes,USE_OLDCMAKECUDA can removed.

**Comment by cjolivier01** [comment]
why can it be removed?

**Comment by yajiedesign** [comment]
It is now possible to automatically detect the version of the cmake to determine which scheme to use.

**Comment by cjolivier01** [comment]
that’s not why it is there. it’s there to not use the built in  cuda build generator

**Comment by yajiedesign** [comment]
It can also maintain the status quo. It doesn't affect the compilation. Only CMAKE_GENERATOR_TOOLSET doesn't work. CMAKE_GENERATOR_TOOLSET needs to be set before project. Otherwise, it is undefined behavior.

**Comment by cjolivier01** [comment]
i don’t understand the statement about status quo?
with your change, is it still possible to build the “old way” without the generator? i see that FIRST_CUDA is no longer turned off after your change.

**Comment by yajiedesign** [comment]
If cmake is lower than 3.9.0, the “old way”  will be automatically enabled.

**Comment by yajiedesign** [comment]
ok,I've updated it again..
at present:
1.if cmake is lower than 3.9.0, auto choose "old way”.
2.if cmake  greater than equal to or equal to 3.9.0 and USE_OLDCMAKECUDA = off,use FIRST CUDA.
3.if USE_OLDCMAKECUDA  = on,use "old way”.

**Comment by cjolivier01** [comment]
We must have the ability to turn it off for any version of cmake, because it causes problems for some use cases such as running unit tests from clion

**Comment by yajiedesign** [comment]
i am updated .

File: CMakeLists.txt [hunk 1]
```diff
@@ -44,16 +44,16 @@ if(USE_CUDA AND NOT USE_OLDCMAKECUDA)
         OR (${CMAKE_GENERATOR} STREQUAL "Unix Makefiles")
       ) AND (
         (${CMAKE_VERSION} VERSION_GREATER "3.9.0") OR (${CMAKE_VERSION} VERSION_EQUAL "3.9.0")
-      )
+      ) AND NOT USE_OLDCMAKECUDA
```
Comment (cjolivier01) [inline CMakeLists.txt]: I think it makes more sense up top in the old place because this else block just reset it again to false, right?

Comment (yajiedesign) [reply CMakeLists.txt]: else block set "FIRST_CUDA" to false,not set "USE_OLDCMAKECUDA".

Once the FIRST_CUDA is set up, it doesn't need to be checked USE_OLDCMAKECUDA

Comment (yajiedesign) [reply CMakeLists.txt]: It can also be in the old place..

**Comment by CodingCat** [comment]
Hi, the community has passed to vote about associating the code changes with JIRA (https://lists.apache.org/thread.html/ab22cf0e35f1bce2c3bf3bec2bc5b85a9583a3fe7fd56ba1bbade55f@%3Cdev.mxnet.apache.org%3E)

We have updated the guidelines for contributors in https://cwiki.apache.org/confluence/display/MXNET/Development+Process, please ensure that you have created a JIRA at https://issues.apache.org/jira/projects/MXNET/issues/ to describe your work in this pull request and include the JIRA title in your PR as [MXNET-xxxx] your title where MXNET-xxxx is the JIRA id

Thanks!

**Comment by marcoabreu** [comment]
@lebeg

**Comment by chinakook** [comment]
In VS2017 with vc 14.0(default compiler of VS2015 but shipped with VS2017) included(cmake -T v140). USE_OLDCMAKECUDA is needed to compile mxnet with cuda.

**Comment by lebeg** [comment]
I think this is indeed a very good change. I am sure that there are alternatives to hardcoding things into CMakeLists.txt with documented dangers of undefined behaviour. Checkout another improvement from my side https://github.com/apache/incubator-mxnet/pull/10564

**Comment by szha** [comment]
ping

**Comment by lebeg** [comment]
There might be additional changes needed to make CUDA usable with cmake https://github.com/apache/incubator-mxnet/issues/12107

**Comment by lupesko** [comment]
This seems to have been abandoned by the author, request for changes not addressed since May 21 :(
@sandeep-krishnamurthy @nswamy @szha suggest we close this PR due to inactivity.

**Comment by sandeep-krishnamurthy** [comment]
@yajiedesign - Please reopen the PR or let us know how to take this forward.
Closing it for now due to inactivity.
Thanks.

**Comment by yajiedesign** [comment]
sorry for later,this pr has expire.

**Pull request closed without merging**